src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -58,21 +60,30 @@ const ProgressiveProfiling = (props) => {
       })
       .catch(() => {});
 
-    let userEnrollmentAction = false;
     if (registrationResponse) {
       setRegistrationResult(registrationResponse);
       sendPageEvent('login_and_registration', 'welcome');
-
-      const queryParams = getAllPossibleQueryParams(registrationResponse.redirectUrl);
-      if ('enrollment_action' in queryParams) {
-        userEnrollmentAction = true;
-      }
     }
+  }, [DASHBOARD_URL, registrationResponse]);
 
-    if (enablePersonalizedRecommendations && !userEnrollmentAction) {
-      setShowRecommendationsPage(true);
+  useEffect(() => {
+    let queryParams = {};
+    let timer = null;
+    if (registrationResponse) {
+      queryParams = getAllPossibleQueryParams(registrationResponse.redirectUrl);
+      if (enablePersonalizedRecommendations && !('enrollment_action' in queryParams)) {
```

Comment:
We need to add cookie logic for testing on stage.

Reply:
we will testing on stage without cookie logic.